depthai_sdk/src/depthai_sdk/components/nn_component.py

```diff
@@ -126,37 +136,63 @@ def _update_device_info(self, pipeline: dai.Pipeline, device: dai.Device, versio
-            # Crop the high-resolution frames so it matches object detection frame aspect ratio
-            # NOTE: objecgt detections have to be set to CROP mode
-            self.imageManip = pipeline.createImageManip()
-            self.imageManip.setResize(*crop)
-            self.imageManip.setMaxOutputFrameSize(crop[0] * crop[1] * 3)
-            self.imageManip.initialConfig.setFrameType(dai.RawImgFrame.Type.BGR888p)
-            self._input._stream_input.link(self.imageManip.inputImage)
-
-            # Create script node, get HQ frames from input.
-            self._multiStageNn = MultiStageNN(pipeline, self._input.node, self.imageManip.out, self._size)
-            self._multiStageNn.configure(self._multi_stage_config)
-            self._multiStageNn.out.link(self.node.input)  # Cropped frames
-            # For debugging, for integral counter
-            self.node.out.link(self._multiStageNn.script.inputs['recognition'])
-            self.node.input.setBlocking(True)
-            self.node.input.setQueueSize(15)
+            crop = int(scale[i] * nn_size[0]), int(scale[i] * nn_size[1])
+            # Crop the high-resolution frames, so it matches object detection frame aspect ratio
+
+            self.image_manip = pipeline.createImageManip()
+            self.image_manip.setNumFramesPool(10)
+            self.image_manip_config = dai.ImageManipConfig()
+            self.image_manip.initialConfig.setFrameType(dai.RawImgFrame.Type.BGR888p)
+
+            self._input._stream_input.link(self.image_manip.inputImage)
+
+            if self._input._is_detector() and self._input._decode_fn is None:
+                self.image_manip.setResize(*crop)
+                self.image_manip.setMaxOutputFrameSize(crop[0] * crop[1] * 3)
+
+                # Create script node, get HQ frames from input.
+                self._multi_stage_nn = MultiStageNN(pipeline, self._input.node, self.image_manip.out, self._size)
+                self._multi_stage_nn.configure(self._multi_stage_config)
+                self._multi_stage_nn.out.link(self.node.input)  # Cropped frames
+
+                # For debugging, for integral counter
+                self.node.out.link(self._multi_stage_nn.script.inputs['recognition'])
+                self.node.input.setBlocking(True)
+                self.node.input.setQueueSize(20)
+            else:
+                # Custom NN
+                self.image_manip.setResize(*self._size)
+                # self.image_manip.setWaitForConfigInput(True)
+                # self.image_manip.inputConfig.setReusePreviousMessage(True)
+
+                self.image_manip.setMaxOutputFrameSize(self._size[0] * self._size[1] * 3)
+
+                self.x_in = pipeline.createXLinkIn()
+                self.x_in.setStreamName("input_queue")
+                self.x_in.setMaxDataSize(frame_size[0] * frame_size[1] * 3)
+                self.x_in.out.link(self.image_manip.inputImage)
```

Comment:
Maybe add TODO for the future that we could also just queue high-res img on the device (script node) then send config from host and Script node would send queued up Image + config to the ImageManip

Reply:
Added TODO